packages/fork-choice/src/protoArray/protoArray.ts

```diff
@@ -214,6 +236,177 @@ export class ProtoArray {
     }
   }
 
+  /**
+   * Optimistic sync validate till validated latest hash, invalidate any decendant branch
+   * if invalidate till hash provided. If consensus fails, this will invalidate entire
+   * forkChoice which will throw on any call to findHead
+   */
+  validateLatestHash(execResponse: LVHExecResponse): void {
+    // Look reverse because its highly likely node with latestValidExecHash is towards the
+    // the leaves of the forkchoice
+    //
+    // We can also implement the index to lookup for exec hash => proto block, but it
+    // still needs to be established properly (though is highly likely) than a unique
+    // exec hash maps to a unique beacon block.
+    // For more context on this please checkout the following conversation:
+    // https://github.com/ChainSafe/lodestar/pull/4182#discussion_r914770167
+
+    if (execResponse.executionStatus === ExecutionStatus.Valid) {
+      const {latestValidExecHash} = execResponse;
+      let latestValidHashIndex = this.nodes.length - 1;
```

Comment:
the default latest valid hash index should not be the last node, maybe make it `undefined` or `null`?

Reply:
thanks for the input, i have initialized to -1, keeping in line with "not found" assignments of the indices. if you still feel strongly for undefined or null to represent `not found`, do respond :slightly_smiling_face: